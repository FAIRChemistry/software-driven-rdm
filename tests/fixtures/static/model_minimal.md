# Module

Some description

## Objects

Other description

### Object

- attribute
  - Type: int
  - Description: Primitive attribute
- multiple_primitive_attribute
  - Type: int[]
  - Description: Multiple primitive attribute
- nested
  - Type: Nested
  - Description: Nested attribute
- inherited_nested
  - Type: InheritedNested
  - Description: Inherited nested attribute

### Nested

- attribute
  - Type: int
  - Description: Primitive attribute

### InheritedNested [*Nested*]

- added_attr
  - Type: int
  - Description: Primitive attribute

## Enumerations

### SomeEnum

```python
VALUE = "value"
```
