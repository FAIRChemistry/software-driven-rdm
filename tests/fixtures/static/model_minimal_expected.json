{
  "objects": [
    {
      "name": "Object",
      "docstring": "",
      "attributes": [
        {
          "name": "attribute",
          "required": false,
          "default": null,
          "type": [
            "int"
          ],
          "description": "Primitive attribute"
        },
        {
          "name": "multiple_primitive_attribute",
          "required": false,
          "type": [
            "int"
          ],
          "description": "Multiple primitive attribute",
          "default_factory": "ListPlus()",
          "multiple": "True"
        },
        {
          "name": "nested",
          "required": false,
          "default": null,
          "type": [
            "Nested"
          ],
          "description": "Nested attribute"
        },
        {
          "name": "inherited_nested",
          "required": false,
          "default": null,
          "type": [
            "InheritedNested"
          ],
          "description": "Inherited nested attribute"
        }
      ],
      "type": "object",
      "subtypes": [],
      "module": "objects"
    },
    {
      "name": "Nested",
      "docstring": "",
      "attributes": [
        {
          "name": "attribute",
          "required": false,
          "default": null,
          "type": [
            "int"
          ],
          "description": "Primitive attribute"
        }
      ],
      "type": "object",
      "subtypes": [],
      "module": "objects"
    },
    {
      "name": "InheritedNested",
      "docstring": "",
      "attributes": [
        {
          "name": "added_attr",
          "required": false,
          "default": null,
          "type": [
            "int"
          ],
          "description": "Primitive attribute"
        }
      ],
      "type": "object",
      "subtypes": [],
      "parent": "Nested",
      "module": "objects"
    }
  ],
  "enums": [
    {
      "name": "SomeEnum",
      "docstring": "",
      "mappings": [
        {
          "key": "VALUE",
          "value": "\"value\""
        }
      ],
      "type": "enum"
    }
  ],
  "inherits": [
    {
      "parent": "Nested",
      "child": "InheritedNested"
    }
  ],
  "compositions": [
    {
      "container": "Object",
      "module": "Nested"
    },
    {
      "container": "Object",
      "module": "InheritedNested"
    }
  ],
  "external_objects": {}
}